### DISCUSSION TITLE
Understanding of i and j in dde.grad jacobian and Hessian

### DISCUSSION THREAD

**Discussion by Shah-Dhruv1** [lululxvi/deepxde#1096] [Q&A]
@lululxvi Thanks for such a amazing library

When i was going through the example of Helmholtz equation in 2D with hole https://deepxde.readthedocs.io/en/latest/demos/pinn_forward/helmholtz.2d.neumann.hole.html while going through the pde part in dy/dxx the value of i was equal to 0 whereas in dy/yy i was equal to 1 
I have gone through the following things  but didn't i still have a doubt:
1) x means the spatial domain value and y here means the concentration(u) but u has only one dimension so why to use i =1 in yy instead of i =0 https://github.com/lululxvi/deepxde/blob/master/deepxde/gradients.py
2) while going through the code and printing shape of u(y) in pde i got shape (none,1) this should give error while using i =1 but it still runned perfectly https://deepxde.readthedocs.io/en/latest/modules/deepxde.html#module-deepxde.gradients 
3)in documentation the part describing i and j are empty
It would be a great help if you could help with it ...
Thanks in advance...

**Comment by lululxvi** [comment]
- Jacobian: https://en.wikipedia.org/wiki/Jacobian_matrix_and_determinant
- Hessian: https://en.wikipedia.org/wiki/Hessian_matrix
